packages/web3-eth-contract/src/contract.ts

```diff
@@ -357,17 +357,15 @@ export class Contract<Abi extends ContractAbi>
 				? optionsOrContextOrReturnFormat
 				: contextOrReturnFormat),
 			provider:
-				typeof addressOrOptionsOrContext !== 'string'
-					? addressOrOptionsOrContext?.provider ??
-					  // eslint-disable-next-line no-nested-ternary
-					  (typeof optionsOrContextOrReturnFormat === 'object' &&
-					  'provider' in optionsOrContextOrReturnFormat
-							? optionsOrContextOrReturnFormat.provider
-							: typeof contextOrReturnFormat === 'object' &&
-							  'provider' in contextOrReturnFormat
-							? contextOrReturnFormat?.provider
-							: Contract.givenProvider)
-					: undefined,
+				(addressOrOptionsOrContext as Web3ContractContext)?.provider ??
+				// eslint-disable-next-line no-nested-ternary
+				(typeof optionsOrContextOrReturnFormat === 'object' &&
+				'provider' in optionsOrContextOrReturnFormat
+					? optionsOrContextOrReturnFormat.provider
+					: typeof contextOrReturnFormat === 'object' &&
+					  'provider' in contextOrReturnFormat
+					? contextOrReturnFormat?.provider
+					: Contract.givenProvider),
```

Comment:
Addressed by [this](https://github.com/web3/web3.js/pull/5669/commits/81ec99b1c71a01a8e6eef24a81458c2b0ec0c880) commit. There was a comment from Nazar saying there was a TypeScript error being thrown by not having the `super` call be the very first thing inside the constructor, but I don't seem to receive such an error

Reply:
Thanks @spacesailor24 ,
This is interesting. And when I search for it, I found that the error when not calling `super ` first is: `A 'super' call must be the first statement in the constructor when a class contains initialized properties or has parameter properties`. 
So, there will be no error if there is no properties initialization in the constructor. And it could be that at some point in time there were some properties initialization that had been deleted later. But the comment was not removed :smile:.